swri_roscpp/include/swri_roscpp/dynamic_parameters.h

```diff
@@ -0,0 +1,450 @@
+#ifndef SWRI_ROSCPP_DYNAMIC_PARAMETERS_H_
+#define SWRI_ROSCPP_DYNAMIC_PARAMETERS_H_
+
+#include <map>
+#include <string>
+
+#include <boost/thread/mutex.hpp>
+
+#include <ros/console.h>
+#include <ros/node_handle.h>
+
+#include <dynamic_reconfigure/Config.h>
+#include <dynamic_reconfigure/ConfigDescription.h>
+#include <dynamic_reconfigure/GroupState.h>
+#include <dynamic_reconfigure/Reconfigure.h>
+
+namespace swri
+{
+  struct DynamicValue
+  {
+    enum Type
+    {
+      Bool = 0,
+      Float = 1,
+      Double = 2,
+      Int = 3,
+      String = 4
+    };
+
+    Type type;
+    std::string name;
+    std::string description;
+
+    void* pointer;//pointer to the parameter to update on change
+
+    // Defaults, maximum and minimum values for this parameter
+    union
+    {
+      double d;
+      bool b;
+      int i;
+    } Default;
+    union
+    {
+      double d;
+      int i;
+    } Min;
+    union
+    {
+      double d;
+      int i;
+    } Max;
+    std::string default_string;// to get around union issues with strings
+  };
+
+  class DynamicParameters
+  {
+    ros::Publisher descr_pub_;
+    ros::Publisher update_pub_;
+    ros::ServiceServer set_service_;
+    ros::NodeHandle nh_;
+
+    std::map<std::string, DynamicValue> values_;
+
+    boost::mutex mutex_;
+
+    bool setConfigCallback(dynamic_reconfigure::Reconfigure::Request &req,
+          dynamic_reconfigure::Reconfigure::Response &rsp)
+    {
+      ROS_INFO("Got configuration change request");
+
+      boost::mutex::scoped_lock lock(mutex_);
+      
+      // update the parameters
+      for (int i = 0; i < req.config.doubles.size(); i++)
+      {
+        dynamic_reconfigure::DoubleParameter param = req.config.doubles[i];
+        std::map<std::string, DynamicValue>::iterator iter = values_.find(param.name);
+        if (iter == values_.end())
+        {
+          ROS_ERROR("Could not find parameter '%s'", param.name.c_str());
+          continue;
+        }
+
+        if (iter->second.type != DynamicValue::Double && iter->second.type != DynamicValue::Float)
+        {
+          ROS_ERROR("Value '%s' was not a double type.", param.name.c_str());
+          continue;
+        }
+
+        if (iter->second.type == DynamicValue::Double)
+        {
+          double* v = (double*)iter->second.pointer;
```

Comment:
Any reason to use C-style casts here rather than `static_cast`s?

Reply:
For all of the cases here, I'd prefer to use `static_cast` so that it's explicit what type of casting you're doing and also because the compiler is better at catching errors if you're using it wrong.  (see http://www.stroustrup.com/bs_faq2.html#static-cast for more details)